kafka-connect-aws-s3/src/main/scala/io/lenses/streamreactor/connect/aws/s3/formats/CsvFormatStreamReader.scala

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2020 Lenses.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.lenses.streamreactor.connect.aws.s3.formats
+
+import java.io.{InputStream, InputStreamReader}
+
+import au.com.bytecode.opencsv.CSVReader
+import io.lenses.streamreactor.connect.aws.s3.model.{BucketAndPath, CsvSourceData}
+
+import scala.util.Try
+
+class CsvFormatStreamReader(inputStreamFn: () => InputStream, bucketAndPath: BucketAndPath, readHeaders: Boolean) extends S3FormatStreamReader[CsvSourceData] {
+
+  private val inputStreamReader = new InputStreamReader(inputStreamFn())
+  private val csvReader = new CSVReader(inputStreamReader)
+  private val iterator = new CsvReaderIteratorAdaptor(csvReader)
+
+  private var columnHeaders: Option[Array[String]] = None
+  private var lineNumber: Long = -1
+
+  private def inventColumnHeaders(size: Int): Option[Array[String]] = Some((1 to size).map(n => s"col$n").toArray)
+
+  override def hasNext: Boolean = iterator.hasNext
+
+  def discoverColumnHeaders(): Option[Array[String]] = {
+
+    if (readHeaders && iterator.hasNext) {
+      lineNumber += 1
+      Some(iterator.next())
+    } else {
+      iterator.peek match {
+        case Some(value) => inventColumnHeaders(value.length)
```

Comment:
If headers are present, we do not read them - it's just skip.  otherwise it's a case of building a structure using the CSV headers as fields and all the values are string (we cannot infer type optimally).

we should never stream the headers/columns. a client would have no clue how to distinguish a record for headers and one for an entry. and even worst if you have multiple producers to a topic with interleaved records

Reply:
CSV parsing has now been removed.
Headers are no longer read.